ocs_ci/ocs/longevity.py

```diff
@@ -514,6 +543,144 @@ def validate_obcs_in_kube_job_reached_running_state(
+            master_describe_dict_out = get_node_resource_utilization_from_oc_describe(
+                node_type="master", print_table=True
+            )
+            oc_describe_res_util_list.append(master_describe_dict_out)
+            worker_describe_dict_out = get_node_resource_utilization_from_oc_describe(
+                node_type="worker", print_table=True
+            )
+            oc_describe_res_util_list.append(worker_describe_dict_out)
+            cluster_sanity_check_dict["resource_utilization"][
+                "oc_describe_nodes_res_util"
+            ] = oc_describe_res_util_list
+
+            # Get the resource utilization of the pods in openshift-storage namespace
+            pod_raw_adm_out = pod_resource_utilization_raw_output_from_adm_top()
+            cluster_sanity_check_dict["resource_utilization"][
+                "adm_top_pods_res_util"
+            ] = pod_raw_adm_out
+
+        if disk_utilization:
+            osd_disk_utilization_list = []
+            cluster_sanity_check_dict["disk_utilization"] = {}
+            # Get OSD utilization
+            osd_filled_dict = get_osd_utilization()
+            osd_disk_utilization_list.append(f"OSD disk utilization: {osd_filled_dict}")
+            log.info(f"OSD Utilization: {osd_filled_dict}")
+            cluster_sanity_check_dict["disk_utilization"][
+                "osd_disk_utilization"
+            ] = osd_filled_dict
+            # Get the percentage of the total used capacity in the cluster
+            total_used_capacity = get_percent_used_capacity()
+            log.info(
+                f"The percentage of the total used capacity in the cluster: {total_used_capacity}"
+            )
+            cluster_sanity_check_dict["disk_utilization"][
+                "cluster_total_used_capacity"
+            ] = total_used_capacity
+
+        log.info("Completed Cluster Sanity Checks")
+
+        return cluster_sanity_check_dict
+
+    def collect_cluster_sanity_checks_outputs(self, dir_name=None):
+        """
+        Collect cluster sanity checks outputs and store the outputs in ocs-ci log directory
+
+        Args:
+            dir_name (str):  By default the cluster sanity checks outputs are stored in
+            ocs-ci-log_path/cluster_sanity_outputs directory.
+            when dir_name input is provided, a new directory with that name gets created under
+            ocs-ci-log_path/cluster_sanity_outputs/<dir_name> and the outputs gets collected inside it
+
+        """
+        destination_dir = (
+            f"{self.cluster_sanity_check_dir}/{dir_name}"
+            if dir_name
+            else self.cluster_sanity_check_dir
+        )
+        if not os.path.isdir(destination_dir):
+            Path(destination_dir).mkdir(parents=True, exist_ok=True)
+        cluster_sanity_out_dict = self.cluster_sanity_check()
+        for key1 in cluster_sanity_out_dict:
+            fopen = open(f"{destination_dir}/{key1}", "w")
```

Comment:
You can use:
with open(f"{destination_dir}/{key1}", 'w') as f:
for key2, value in cluster_sanity_out_dict[key1].items():
                f.write(f"{key2} : {value}\n")
and spare the close

Reply:
Done